### PR TITLE
homework 4

### PR DESCRIPTION
**Pull request by Lubovgroma**

### CODE REVIEW

File: homework4/task 3.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+movieName: 'Predator',
+releaseYear: 1987,
+directedBy: 'McTiernan',
+runningTimeInMinutes: 107,
+},
+];
+
+console.log(movies.sort(byProperty('releaseYear', '>'))); // виведе масив фільмів посортованих по року випуску, від старішого до новішого*
+console.log(movies.sort(byProperty('runningTimeInMinutes', '<'))); // виведе масив фільмів посортованих по їх тривалості, від найдовшого до найкоротшого*
+console.log(movies.sort(byProperty('movieName', '>'))); // виведе масив фільмів посортованих по назві, в алфавітному порядку*
```
Comment (sharkich) [inline homework4/task 3.js]: do not use function before declaration

File: homework4/task 3.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                    return negativeValue;
+                }
+                if (a > b) {
+                    return positiveValue;
+                }
+                return equalValue;
+            }
+            
+        }
+    }
```
Comment (sharkich) [inline homework4/task 3.js]: you can improve the solution is you try to use `a[property]`

**Review by sharkich — CHANGES_REQUESTED** [review]

File: homework4/task1-2.js [hunk 3]
```diff
@@ -2,23 +2,22 @@
 
 //---------------task 1------------------
 
+function addThemAll (...allArg ){
+    return allArg.reduce((a, b)=>a+b);
```
Comment (Lubovgroma) [inline homework4/task1-2.js]: I have tried to add default values like that --  allArg.reduce((a = 1, b = 1)=>a+b); , but error occurs "Reduce of empty array with no initial value"

File: homework4/task 3.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+},
+];
+
+function byProperty(property, direction) {
+    const negativeValue = -1;
+    const equalValue = 0;
+        if(direction === '>')
+        {
+            return (x,y) => {
+                if ( y[property] > x[property]){
```
Comment (Lubovgroma) [inline homework4/task 3.js]: used x[property]

File: homework4/task 4.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+//---------------task 4------------------
+
+// with setInterval
+function detonatorTimer1(delay){
+    let timer = delay;
+    let timerId = setInterval(() => {
+        if (timer === 0) {
+            clearInterval(timerId);
```
Comment (Lubovgroma) [inline homework4/task 4.js]: clearInterval used inside the function

File: homework4/task 4.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+// with setTimeout
+function detonatorTimer2(delay){
+    let timer = delay;
+    let timerId = setTimeout(() => {
+        if (timer === 0) {
+            return console.log("BOOM!");
+        }
+        console.log(timer);
+        return detonatorTimer2(timer-1);
```
Comment (Lubovgroma) [inline homework4/task 4.js:25]: Recursion used

File: homework4/task5-6-7.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+let securedAircraftName = airline.aircraftName.bind(airline);
+let securedCountryAndHub = airline.countryAndHub.bind(airline);
+
+setTimeout(securedAircraftName, 1000); // виведе коректний результат*
+setTimeout(securedCountryAndHub, 2000); // виведе коректний результат*
+
+//--------------task 7-------------------
+
+function someFunction (...arg){
```
Comment (Lubovgroma) [inline homework4/task5-6-7.js:34]: function accepts any amount of arguments

File: homework4/task 3.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+runningTimeInMinutes: 107,
+},
+];
+
+function byProperty(property, direction) {
+    const negativeValue = -1;
+    const equalValue = 0;
+        if(direction === '>')
+        {
+            return (x,y) => {
```
Comment (sharkich) [inline homework4/task 3.js]: Change logic

File: homework4/task 4.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+//---------------task 4------------------
+
+// with setInterval
+function detonatorTimer1(delay){
+    let timer = delay;
+    let timerId = setInterval(() => {
+        if (timer === 0) {
+            clearInterval(timerId); 
+            return console.log('BOOM!');
```
Comment (sharkich) [inline homework4/task 4.js]: 2 lines

File: homework4/task 4.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+//---------------task 4------------------
+
+// with setInterval
+function detonatorTimer1(delay){
+    let timer = delay;
+    let timerId = setInterval(() => {
```
Comment (sharkich) [inline homework4/task 4.js]: const

File: homework4/task 4.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        }
+        console.log(timer);
+        timer--;
+    }, 1000);
+}
+
+// with setTimeout
+function detonatorTimer2(delay){
+    let timer = delay;
+    let timerId = setTimeout(() => {
```
Comment (sharkich) [inline homework4/task 4.js]: const

File: homework4/task 4.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        if (timer === 0) {
+            return console.log("BOOM!");
```
Comment (sharkich) [inline homework4/task 4.js]: 2 lines

File: homework4/task1-2.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+//---------------task 1------------------
+
+function addThemAll (...allArg ){
+    return allArg.reduce((a, b)=>a+b);
```
Comment (sharkich) [inline homework4/task1-2.js]: defasult

**Review by sharkich — CHANGES_REQUESTED** [review]

File: homework4/task 4.js [hunk 14]
```diff
@@ -5,9 +5,8 @@ function detonatorTimer1(delay){
     let timer = delay;
     const timerId = setInterval(() => {
         if (timer === 0) {
-            clearInterval(timerId); 
-            return 
             console.log('BOOM!');
+            return clearInterval(timerId);
```
Comment (Lubovgroma) [inline homework4/task 4.js:9]: can I return clearInterval(timerId) ?

File: homework4/task 4.js [hunk 15]
```diff
@@ -18,9 +17,9 @@ function detonatorTimer1(delay){
 function detonatorTimer2(delay){
     let timer = delay;
     const timerId = setTimeout(() => {
-        if (timer === 0) {
-            return 
+        if (timer === 0) { 
             console.log("BOOM!");
+            return 1;
```
Comment (Lubovgroma) [inline homework4/task 4.js:22]: added return 1 just to stop recursion, is it ok?

**Review by Lubovgroma — COMMENTED** [review]
Please check only last updated version: 
[homework 4/last version] (https://github.com/Lubovgroma/Prjctr-course/pull/5/commits/76911b4fa0ad484d88b591da4b96ca6979a2c849)

**Pull request merged**